internal/cri/io/container_io.go

```diff
@@ -211,15 +212,40 @@ func (c *ContainerIO) Attach(opts AttachOptions) {
 	}
 
 	if opts.Stdout != nil {
-		wg.Add(1)
 		wc, close := cioutil.NewWriteCloseInformer(opts.Stdout)
 		c.stdoutGroup.Add(stdoutKey, wc)
+		wg.Add(2)
+		// Send an empty data to check stream is closed
+		go func() {
```

Comment:
```suggestion
                wg.Add(1)
		go func() {
```

Reply:
done,  my first pr https://github.com/ningmingxiao/containerd/pull/14/files#diff-66244a50c152fd2b44ae06533fd31001c97caef29b7d4b5e0c0492473158a4e4

```
	ctx.conn = conn
	ctx.closeChan = conn.CloseChan()
```
I can use this chan to know spdy strem is closed, but now these code move to k8s and spy will be deprecated.
add a for func is not perfect.

Reply:
nod let's bring in the cavalry and get a plan together...